test/e2e/certmanager/certmanager_test.go

```diff
@@ -0,0 +1,231 @@
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/utils/ptr"
+
+	"sigs.k8s.io/kueue/apis/config/v1beta1"
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+	workloadappwrapper "sigs.k8s.io/kueue/pkg/controller/jobs/appwrapper"
+	workloadjob "sigs.k8s.io/kueue/pkg/controller/jobs/job"
+	workloadjobset "sigs.k8s.io/kueue/pkg/controller/jobs/jobset"
+	workloadlws "sigs.k8s.io/kueue/pkg/controller/jobs/leaderworkerset"
+	"sigs.k8s.io/kueue/pkg/util/testing"
+	awtesting "sigs.k8s.io/kueue/pkg/util/testingjobs/appwrapper"
+	testingjob "sigs.k8s.io/kueue/pkg/util/testingjobs/job"
+	testingjobset "sigs.k8s.io/kueue/pkg/util/testingjobs/jobset"
+	testinglws "sigs.k8s.io/kueue/pkg/util/testingjobs/leaderworkerset"
+	"sigs.k8s.io/kueue/test/util"
+)
+
+var _ = ginkgo.Describe("CertManager", ginkgo.Ordered, func() {
+	var (
+		ns           *corev1.Namespace
+		defaultRf    *kueue.ResourceFlavor
+		localQueue   *kueue.LocalQueue
+		clusterQueue *kueue.ClusterQueue
+	)
+
+	ginkgo.BeforeAll(func() {
+		configurationUpdate := time.Now()
+		config := defaultKueueCfg.DeepCopy()
+		config.InternalCertManagement = &v1beta1.InternalCertManagement{
+			Enable: ptr.To(false),
+		}
+		util.ApplyKueueConfiguration(ctx, k8sClient, config)
+		util.RestartKueueController(ctx, k8sClient)
+		ginkgo.By("Checking Kueue certificates are ready", func() {
+			util.WaitForCertificateReady(ctx, k8sClient,
+				"serving-cert", "kueue-system", 5*time.Minute)
+		})
+		ginkgo.By("Checking webhook CA bundles are injected", func() {
+			util.WaitForWebhookCABundleReady(ctx, k8sClient, 2*time.Minute)
+		})
+		util.WaitForKueueAvailability(ctx, k8sClient)
+		ginkgo.GinkgoLogr.Info("Kueue configuration updated", "took", time.Since(configurationUpdate))
+	})
+
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{GenerateName: "e2e-cert-manager-"},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+
+		defaultRf = testing.MakeResourceFlavor("default").Obj()
+		gomega.Expect(k8sClient.Create(ctx, defaultRf)).Should(gomega.Succeed())
+
+		clusterQueue = testing.MakeClusterQueue("cluster-queue").
+			ResourceGroup(*testing.MakeFlavorQuotas(defaultRf.Name).
+				Resource(corev1.ResourceCPU, "2").
+				Resource(corev1.ResourceMemory, "2G").Obj()).Obj()
+		gomega.Expect(k8sClient.Create(ctx, clusterQueue)).Should(gomega.Succeed())
+
+		localQueue = testing.MakeLocalQueue("main", ns.Name).ClusterQueue("cluster-queue").Obj()
+		gomega.Expect(k8sClient.Create(ctx, localQueue)).Should(gomega.Succeed())
+	})
+
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+		util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, clusterQueue, true, util.LongTimeout)
+		util.ExpectObjectToBeDeletedWithTimeout(ctx, k8sClient, defaultRf, true, util.LongTimeout)
+	})
+
+	ginkgo.When("CertManager is Enabled", func() {
+		ginkgo.It("should admit a Job", func() {
```

Comment:
So you are missing quite a few integrations..
I don't see pod, statefulset, deployment, leaderworkerset, and all the kubeflow ones.

I was thinking that maybe we could consider running the same e2e test suite but with just cert manager kueue.

I think these tests are useful as is but they don't cover all the other CRDs that Kueue has.

Reply:
WDYT @tenzen-y @mimowo?

Reply:
I want to have only Job test (current approach) since this E2E verifies contacting between Kueue and CertManager. Each integration does not depend on Certificate Management Mechanism.

Reply:
I'm not sure about the end-state, but the PR is already large, so +1 on focusing on testing for Job. We can extend the coverage in the future if we have evidence it is not sufficient.